ckks/bootstrapping/parameters_literal.go

```diff
@@ -0,0 +1,339 @@
+//	For example: [][]int{{45}, {46}, {47}} means that the CoeffsToSlots step will use three levels, each with one prime. Primes are consumed in reverse order,
+//	so in this example the first matrix will use the prime of 47 bits, the second the prime of 46 bits, and so on.
+//	Non standard parameterization can include multiple scaling factors for a same prime, for example [][]int{{30}, {30, 30}} will use two levels for three matrices.
+//	The first two matrices will consume a prime of 30 + 30 bits, and have a scaling factor which prime^(1/2), and the third matrix will consume the second prime of 30 bits.
+//
+// SlotsToCoeffsFactorizationDepthAndLogScales: the scaling factor and distribution of the moduli for the CoeffsToSlots (homomorphic decoding) step.
+//
+//	Parameterization is identical to C2SLogScale. and the default value is [][]int{{39}, {39}, {39}}
+//
+// EvalModLogScale: the scaling factor used during the EvalMod step (all primes will have this bit-size).
+//
+//	Default value is 60.
+//
+// EphemeralSecretWeight: the Hamming weight of the ephemeral secret, by default set to 32, which ensure over 128-bit security for an evaluation key of modulus 121 bits.
+//
+//	The user can set this value to 0 to use the regular bootstrapping circuit without the ephemeral secret encapsulation.
+//	Be aware that doing so will impact the security, precision, and failure probability of the bootstrapping circuit.
+//	See https://eprint.iacr.org/2022/024 for more information.
+//
+// LogMessageRatio: the log of expected ratio Q[0]/|m|, by default set to 8 (ratio of 256.0).
+//
+//		This ratio directly impacts the precision of the bootstrapping.
+//		The homomorphic modular reduction x mod 1 is approximated with by sin(2*pi*x)/(2*pi), which is a good approximation
+//		when x is close to the origin. Thus a large message ratio (i.e. 2^8) implies that x is small with respect to Q, and thus close to the origin.
+//		When using a small ratio (i.e. 2^4), for example if ct.Scale is close to Q[0] is small or if |m| is large, the ArcSine degree can be set to
+//	 a non zero value (i.e. 5 or 7). This will greatly improve the precision of the bootstrapping, at the expense of slightly increasing its depth.
+//
+// SineType: the type of approximation for the modular reduction polynomial. By default set to advanced.CosDiscret.
+//
+// K: the range of the approximation interval, by default set to 16.
+//
+// SineDeg: the degree of the polynomial approximation of the modular reduction polynomial. By default set to 30.
+//
+// DoubleAngle: the number of double angle evaluation. By default set to 3.
+//
+// ArcSineDeg: the degree of the ArcSine Taylor polynomial, by default set to 0.
+type ParametersLiteral struct {
+	CoeffsToSlotsFactorizationDepthAndLogScales [][]int           // Default: [][]int{{56}, {56}, {56}, {56}}
```

Comment:
idea: 
- move this field (and the SlotToCoeffs) one could be in the `HomomorphicDFTMatrixLiteral` struct
- move the logic to generate the matrices parameters from `[][]int` in the homomorphic DFT constructor, the constructor checks if the level+moduli are specified, if not it revert to the normal construction from the other fields.
- have `ParametersLiteral` accept `HomomorphicDFTMatrixLiteral` as `CoeffToSlotParameter` field.

This way:
1. you have a user-friendlier constructor of encoding matrices in the advanced package
2. the bootstrapping is more parameterizable because it no longer hides the encoding matrices parameters
3. slightly better readability of the literals with something like:
```
ParametersLiteral{
   SlotsToCoeffsParams: HomomorphicDFTMatrixLiteral{FromFactorization: [][]int{{42}, {42}, {42}}} 
   ...
}
```

Not sure this is possible but it seems like the right thing to do.

Reply:
I get the reasoning, but this is not possible. The `HomomorphicDFTMatrixLiteral` relies on a moduli chain that already exists. However this is not the case when creating the bootstrapping parameters. These additional parameters are needed there because the moduli chain for the bootstrapping circuit needs to be created. So they are not for the  `HomomorphicDFTMatrixLiteral` but for the creation of the moduli chain of the bootstrapping circuit.

The constructor for the `HomomorphicDFT` matrices in the `advanced` package is already very user friendly, it is difficult to do better. Here is an example:

```
// CoeffsToSlotsParameters homomorphic decoding parameters
var CoeffsToSlotsParameters = ckksAdvanced.HomomorphicDFTMatrixLiteral{
  Type:       ckksAdvanced.Encode,
  LogSlots:   LogSlots,
  LevelStart: 1,
  Levels:     []int{1},
}
```

Besides, if a user really want to give a custom linear transform, it is possible, since the fields in the `bootstrapping.Parameters` are public. So the option for more customization is there.

Reply:
Got it. Thanks!